Use xplat/caffe2:ovrsource_aten_Config.h for arvr build modes

tarun292 opened the pull request:
Summary: For arvr modes we should use `//xplat/caffe2:ovrsource_aten_Config.h` to get the ATen config headers as the `ovrsource` variants are what's used in arvr repos and using ``"//xplat/caffe2:ovrsource_aten_Config.h"` causes the ovrsource ones to get overwritten and hence results in build failures.

Differential Revision: D71506180

swolchok commented on runtime/core/portable_type/c10/c10/targets.bzl:
```diff
@@ -78,9 +78,8 @@ def define_common_targets():
-            "//xplat/caffe2:generated_aten_config_header",
             "//xplat/caffe2/c10:c10_headers",
-        ],
+        ] + ["//xplat/caffe2:ovrsource_aten_Config.h" if is_arvr_mode() else "//xplat/caffe2:generated_aten_config_header",],
```
do we also need to cover line 63 above?

tarun292 replied:
shouldn't be needed as arvr only uses xplat_exported_deps and not fbcode_exported_deps

tarun292 committed:
Use xplat/caffe2:ovrsource_aten_Config.h for arvr build modes (#9462)

Summary:

For arvr modes we should use `//xplat/caffe2:ovrsource_aten_Config.h` to get the ATen config headers as the `ovrsource` variants are what's used in arvr repos and using ``"//xplat/caffe2:ovrsource_aten_Config.h"` causes the ovrsource ones to get overwritten and hence results in build failures.

Reviewed By: thejarlid

Differential Revision: D71506180
